include/platform/messagepool.h

```diff
@@ -63,14 +63,13 @@ extern "C" {
 #endif
 
 /**
- * Initialize the platform implemented message pool with the provided buffer list.
+ * Initialize the platform implemented message pool.
  *
- * @param[in] aFreeBufferList   A linked list of free buffers.
- * @param[in] aNumFreeBuffers   A pointer to an int containing the number of free buffers in aFreeBufferList.
- * @param[in] aBufferSize       The size in bytes of a Buffer object.
+ * @param[in] aMinNumFreeBuffers   An int containing the minimum number of free buffers desired by OpenThread.
+ * @param[in] aBufferSize          The size in bytes of a Buffer object.
  *
  */
-void otPlatMessagePoolInit(struct BufferHeader *aFreeBufferList, int *aNumFreeBuffers, size_t aBufferSize);
+void otPlatMessagePoolInit(int aMinNumFreeBuffers, size_t aBufferSize);
```

Comment:
A solution that is implemented with dynamic allocation, one where the number of available buffers can't be known should simply return a fixed value that will satisfy the worst case for ReclaimBuffers.  As you can see, the call to this API is in ReclaimBuffers which is used to resize a message.  The code was written to learn whether the buffer could be resized before trying to resize it.  Changing the code so that it doesn't do that is a separate pull-request I think.